### ISSUE TITLE
Add option to control logging verbosity

### ISSUE THREAD

**Issue by arnikz** [PBR/QTM#51]
See a nice tutorial [here](https://www.loggly.com/ultimate-guide/java-logging-basics/).

**Comment by matthijsbrouwer** [comment]
Implemented log4j logger, output to console limited to
```
 INFO	2019-07-15 11:09:55,433	Solr server has been restarted.
 INFO	2019-07-15 11:10:06,684	===============
 INFO	2019-07-15 11:10:06,684	QTLTableMiner++
 INFO	2019-07-15 11:10:06,684	===============
 INFO	2019-07-15 11:10:06,684	Input List: [PMC4321030, PMC4266912]
 INFO	2019-07-15 11:10:06,767	- Processing article PMC4321030
 INFO	2019-07-15 11:10:07,989	- Processing article PMC4266912
 INFO	2019-07-15 11:10:08,029	Solr server has been restarted.
 INFO	2019-07-15 11:10:19,326	Inserting entries to the database
 INFO	2019-07-15 11:10:20,301	Searching QTL in tables
 INFO	2019-07-15 11:10:21,130	Writing results into qtl.csv
 INFO	2019-07-15 11:10:21,141	=======
 INFO	2019-07-15 11:10:21,141	RESULTS
 INFO	2019-07-15 11:10:21,141	=======
 INFO	2019-07-15 11:10:21,141	Total number of processed articles: 2
 INFO	2019-07-15 11:10:21,141	Total number of trait tables: 9
 INFO	2019-07-15 11:10:21,141	Total number of QTL statements: 41
 INFO	2019-07-15 11:10:21,142	SQLite file: qtl.db
 INFO	2019-07-15 11:10:21,142	CSV file: qtl.csv
 INFO	2019-07-15 11:10:21,170	Memory used (KB): 6769
 INFO	2019-07-15 11:10:21,170	Total runtime (HH:MM:SS): 00:00:25
 INFO	2019-07-15 11:10:21,170	Solr server has been stoped.
```
Full log in logfile `QTM.log`, configuration logging in `log4j.properties`.

**Comment by arnikz** [comment]
@matthijsbrouwer: It works fine. Could you add `(default: 4 [INFO])` to the help message?

**Comment by matthijsbrouwer** [comment]
Changed:
- Verbose argument only available if a console handler is defined in the log4j.properties
- Derive and show the default option also from this file

```
#appender console
log4j.appender.console=org.apache.log4j.ConsoleAppender
log4j.appender.console.layout=org.apache.log4j.PatternLayout
log4j.appender.console.layout.ConversionPattern=%5p\t%d\t%m%n
log4j.appender.console.Threshold=INFO
```

```
  -V VERBOSE, --verbose VERBOSE
                         verbosity console output: 0-7 for OFF, FATAL, ERROR, WARN, INFO, DEBUG, TRACE or ALL (default: 4 [INFO])
```

**Comment by arnikz** [comment]
@matthijsbrouwer: I've tried the commit https://github.com/PBR/QTM/commit/88e12bd79cb328d3e78042a95ec038a70181d7bc but the `-V` arg is shown regardless of the `log4j.properties` file (which is not included in the commit). Actually, this could be a preferred behaviour and one could override the default verbosity (as defined in this file) via CLI. Btw. Where in the source are the properties read to configure the logger?

**Comment by matthijsbrouwer** [comment]
log4j.properties was already added in https://github.com/PBR/QTM/commit/3839f26905015ab323e32a7b906ad1d41c562f05 when log4j logging was implemented.

I'm not sure what you tried exactly, but for me it does seem to work fine:

Regarding the question about the location where the properties are read: it is log4j, so somewhere in the log4j source is implemented to look for a log4j.properties file. And the use of log4j in this project is defined in `src/main/java/nl/esciencecenter/qtm/Main.java`. 

A default log4j.properties file is put in `src/main/resources` and in the maven configuration set to be copied to the build path on install. So, after a `mvn install`, in the `target` directory we have not only the QTM jar, but also a copy of the log4j.properties file.

For the testing, start with 
```
mvn clean install
```
Then try default behavior after install for: `./QTM -h`
```
  -V VERBOSE, --verbose VERBOSE
                         verbosity console output: 0-7 for OFF, FATAL, ERROR, WARN, INFO, DEBUG, TRACE or ALL (default: 4 [INFO])
```
The verbosity parameter can be set, and the default level is [INFO] (as configured in the default log4j.properties)

Now, without rebuilding, adjust ./target/log4j.properties, change the Threshold for the defined ConsoleAppender to `DEBUG`, and try again `./QTM -h`
```
  -V VERBOSE, --verbose VERBOSE
                         verbosity console output: 0-7 for OFF, FATAL, ERROR, WARN, INFO, DEBUG, TRACE or ALL (default: 5 [DEBUG])
```
Still the verbosity parameter can be set, but the default level has changed to [DEBUG].

Finally comment out the defined ConsoleAppender in the log4j.properties, and try again `./QTM -h`. You will see that the verbosity parameter has disappeared and can't be used anymore.

**Comment by arnikz** [comment]
You're right about the `log4j.properties` file(s):
```The behaviour as stated in "Verbose argument only available if a console handler is defined in the `log4j.properties`" and not seeing this file in the root was not expected.
find QTM -name log4j.properties
QTM/src/main/resources/log4j.properties
QTM/solr/core/log4j.properties
QTM/solr/server/resources/log4j.properties
QTM/solr/server/scripts/cloud-scripts/log4j.properties
QTM/solr/example/resources/log4j.properties
```
Are all these needed? One for QTM and one for Solr should suffice, right?

I was expecting in the `src/main/java/nl/esciencecenter/qtm/Main.java` something like
`PropertyConfigurator.configure("log4j.properties");`

as done here

https://github.com/PBR/QTM/blob/88e12bd79cb328d3e78042a95ec038a70181d7bc/src/main/java/nl/esciencecenter/qtm/Main.java#L61

Given the description "Verbose argument only available if a console handler is defined in the `log4j.properties`" and not seeing this file in the root, the behaviour was not expected (or unclear) for me but it works fine as suggested with more details;-)

**Comment by arnikz** [comment]
Fixed in 88e12bd79cb328d3e78042a95ec038a70181d7bc.
